subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/KotlinBuildScriptIntegrationTest.kt

```diff
@@ -119,6 +119,7 @@ class KotlinBuildScriptIntegrationTest : AbstractKotlinIntegrationTest() {
             }
 
             task("test") {
+                doNotTrackState("no state")
```

Comment:
Why is this necessary now?

Reply:
This is my guinea pig for trying out if source pinning works, since the compilation will fail if I use a Gradle version older than 7.3.

Reply:
I removed the changes to the tests.